### PR TITLE
Add required field into template parameter

### PR DESCRIPTION
**Pull request by JohnNiang**
### What type of PR is this?

/kind feature

### What this PR does / why we need it:

As mentioned in the title.

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:

Please check the following list before waiting reviewers:

- [ ] Already committed the CRD files to [the Helm Chart](https://github.com/kubesphere-sigs/ks-devops-helm-chart/) if you created some new CRDs
- [ ] Already [added the permission](https://github.com/kubesphere/ks-installer/blob/9e063b085a0e43fdb3d0d9e3e7f4149146f14b9c/roles/ks-core/prepare/files/ks-init/role-templates.yaml) for the new API
- [ ] Already added the RBAC markers for the new controllers

### Does this PR introduce a user-facing change??

```release-note
None
```

### CODE REVIEW

File: pkg/api/devops/v1alpha3/template_types.go [hunk 1]
```diff
@@ -48,6 +48,10 @@ type TemplateParameter struct {
 	//+optional
 	Description string `json:"description,omitempty"`
 
+	// Required indicates if this parameter is mandatory.
+	//+optional
+	Required bool `json:"required"`
```
Comment (LinuxSuRen) [inline pkg/api/devops/v1alpha3/template_types.go:53]: I'm wondering if this field needs `omitempty` or not. What do you think about it?

Comment (JohnNiang) [reply pkg/api/devops/v1alpha3/template_types.go]: I agree with you. Good catch!

**Review by LinuxSuRen — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**